modules/build/src/main/scala/scala/build/options/BuildOptions.scala

```diff
@@ -168,6 +169,55 @@ final case class BuildOptions(
       }
     }
 
+  def latestSupportedScalaVersion(prefixScalaVersion: Option[String]): coursier.core.Version = {
+
+    // used when downloading is fail
+    lazy val defaultStableScalaVersions = StableScalaVersion(
+      version,
+      Seq(defaultScala212Version, defaultScala213Version, defaultScalaVersion)
+    )
+
+    val supportedScalaVersionsUrl = scalaOptions.scalaVersionsUrl
+
+    val task = {
+      val art = Artifact(supportedScalaVersionsUrl).withChanging(true)
+      finalCache.file(art).run.flatMap {
+        case Left(e) => Task.fail(e)
+        case Right(f) => Task.delay {
+            val content = os.read.bytes(os.Path(f, Os.pwd))
+            content
```

Comment:
Let's read it as a string, and decode it from here (so that we don't need the `Try` below)
```suggestion
            val content = os.read(os.Path(f, Os.pwd))
            upickle.default.read[Seq[StableScalaVersion]](content)
```